gc/base/MarkingScheme.cpp

```diff
@@ -429,6 +429,25 @@ MM_MarkingScheme::fixupForwardedSlotOutline(GC_SlotObject *slotObject) {
 #endif /* OMR_GC_CONCURRENT_SCAVENGER */
 }
 
+void
+MM_MarkingScheme::fixupForwardedSlotOutline(omrobjectptr_t *slotPtr) {
+#if defined(OMR_GC_CONCURRENT_SCAVENGER)
+	bool const compressed = _extensions->compressObjectReferences();
+	if (_extensions->getGlobalCollector()->isStwCollectionInProgress()) {
+		MM_ForwardedHeader forwardHeader(*slotPtr, compressed);
+		omrobjectptr_t forwardPtr = forwardHeader.getNonStrictForwardedObject();
+
+		if (NULL != forwardPtr) {
+			if (forwardHeader.isSelfForwardedPointer()) {
+				forwardHeader.restoreSelfForwardedPointer();
+			} else {
+				*slotPtr =forwardPtr;
+			}
+		}
+	}
+#endif /* OMR_GC_CONCURRENT_SCAVENGER */
+}
+
```

Comment:
can the existing method be rewritten to share to code with this new method to be like this:

```
bool
fixupForwardedSlotOutline(GC_SlotObject *slotObject)
{
	omrobjectptr_t slot = slotObject->readReferenceFromSlot();
	fixupForwardedSlotOutline(env, slot);
	slotObject->writeReferenceToSlot(slot);
}

Reply:
... and possibly move it to hpp and inline it